modules/mauth-signer/src/test/scala/com/mdsol/mauth/SignerProtocolSuiteSpec.scala

```diff
@@ -0,0 +1,84 @@
+package com.mdsol.mauth
+
+import java.security.Security
+import java.util.UUID
+
+import com.mdsol.mauth.test.utils.ProtocolTestSuiteHelper
+import com.mdsol.mauth.util.MAuthKeysHelper.getPrivateKeyFromString
+import com.mdsol.mauth.util.{EpochTimeProvider, MAuthSignatureHelper}
+import org.bouncycastle.jce.provider.BouncyCastleProvider
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.collection.JavaConverters._
+
+class SignerProtocolSuiteSpec extends AnyFlatSpec with Matchers with MockFactory {
+  private val mockEpochTimeProvider = mock[EpochTimeProvider]
+
+  Security.addProvider(new BouncyCastleProvider)
+
+  it should "pass the tests in protocol test suite for MWSV2" in {
+    Option(ProtocolTestSuiteHelper.loadSigningConfig()) match {
+      case Some(signingConfig) =>
+        Option(ProtocolTestSuiteHelper.getTestCases) match {
+          case Some(testCases) =>
+            val uuid = UUID.fromString(signingConfig.getAppUuid)
+            val privateKey = getPrivateKeyFromString(signingConfig.getPrivateKey)
+            val mAuthSigner = new DefaultSigner(uuid, privateKey, mockEpochTimeProvider, true)
+
+            for (caseName <- testCases) {
+              println("Test case: " + caseName)
```

Comment:
We probably want to make each of these an individual test case instead of all tests being run inside one single test so that one failing test does not kill every test after it. Can we split them up?

Reply:
It is good suggestion. I am not sure how to split them up in scala, any suggestion?

Reply:
Here's the basic idea:

```
testCases.foreach { testCase =>
  testCase.description in {
    // run the tests
  }
}
```

Also, if we fail to load something it should abort(fail) the whole test suite, not succeed. Throwing exceptions is fine in this case. Something like 

```
class RequestAuthenticatorProtocolSuiteSpec extends .. {
  val signingConfig = Option(ProtocolTestSuiteHelper.loadSigningConfig()).getOrElse(throw new Exception("Cannot load signing config"))
  val testCases = Option(ProtocolTestSuiteHelper.getTestCases).getOrElse(throw new Exception("Unable to load test cases"))
  // .... etc etc
}

```

Reply:
refactor as suggested in https://github.com/mdsol/mauth-jvm-clients/pull/124/commits/a91f9b5eca383d2e1967deb7470ae8810ba7d76d
At current stage, we only do test if required files are loaded properly, will skip test if the expected files are not available.